File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l=0;
        int r = nums.size()-1;
        int minV = nums[0];
        while(l<=r){
            int m = l + (r-l)/2;
            if(nums[l] < nums[r]){
                minV = min(minV,nums[l]);
                break;
            }
            minV = min(minV,nums[m]);
            if(nums[m]>=nums[l]){
                l = m+1;
            }
            else{
                r = m-1;
            }
        }
        return minV;
    }
};

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-1.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        a = sum(nums)
        if a%2==1:
            return False
        a = a/2
        def dfs(i, rem):
            if rem==0:
                return True
            if i>=len(nums):
                return False
            a = nums[i]
            return dfs(i+1, rem-a) or dfs(i+1, rem)
        return dfs(0,a)


            
            

            

